File: src/main/java/de/moritzf/quota/idea/QuotaDisplayMode.java
package de.moritzf.quota.idea;

import java.util.Locale;

/**
 * Rendering mode for the status bar quota widget.
 */
public enum QuotaDisplayMode {
    ICON_ONLY("Icon only"),
    PERCENTAGE_BAR("Percentage bar"),
    CAKE_DIAGRAM("Cake diagram");

    private final String displayName;

    QuotaDisplayMode(String displayName) {
        this.displayName = displayName;
    }

    public static QuotaDisplayMode fromStorageValue(String value) {
        if (value == null || value.isBlank()) {
            return ICON_ONLY;
        }
        String normalized = value.trim().toUpperCase(Locale.ROOT);
        for (QuotaDisplayMode mode : values()) {
            if (mode.name().equals(normalized)) {
                return mode;
            }
        }
        return ICON_ONLY;
    }

    @Override
    public String toString() {
        return displayName;
    }
}

File: src/main/kotlin/de/moritzf/quota/idea/QuotaDisplayMode.kt
package de.moritzf.quota.idea

/**
 * Rendering mode for the quota indicator.
 */
enum class QuotaDisplayMode(private val displayName: String) {
    ICON_ONLY("Icon only"),
    PERCENTAGE_BAR("Percentage bar"),
    CAKE_DIAGRAM("Cake diagram");

    override fun toString(): String = displayName

    companion object {
        @JvmStatic
        fun supportedFor(location: QuotaIndicatorLocation): List<QuotaDisplayMode> {
            return when (location) {
                QuotaIndicatorLocation.STATUS_BAR -> entries
                QuotaIndicatorLocation.MAIN_TOOLBAR -> listOf(ICON_ONLY, CAKE_DIAGRAM)
            }
        }

        @JvmStatic
        fun sanitizeFor(location: QuotaIndicatorLocation, displayMode: QuotaDisplayMode): QuotaDisplayMode {
            return if (displayMode in supportedFor(location)) displayMode else ICON_ONLY
        }

        @JvmStatic
        fun fromStorageValue(value: String?): QuotaDisplayMode {
            if (value.isNullOrBlank()) {
                return ICON_ONLY
            }

            val normalized = value.trim()
            return entries.firstOrNull { it.name.equals(normalized, ignoreCase = true) } ?: ICON_ONLY
        }
    }
}
